src/datachain/config.py

```diff
@@ -1,62 +1,131 @@
-import os
 from collections.abc import Mapping
-from typing import TYPE_CHECKING, Optional
+from contextlib import contextmanager
+from typing import Optional, Union
 
-from tomlkit import load
+from tomlkit import TOMLDocument, dump, load
 
-if TYPE_CHECKING:
-    from tomlkit import TOMLDocument
+from datachain.utils import DataChainDir, global_config_dir, system_config_dir
 
 
-def read_config(datachain_root: str) -> Optional["TOMLDocument"]:
-    config_path = os.path.join(datachain_root, "config")
-    try:
-        with open(config_path, encoding="utf-8") as f:
-            return load(f)
-    except FileNotFoundError:
-        return None
+class Config:
```

Comment:
Looks good to me, it would be great to add tests for this class.

Reply:
Added tests and fixes in https://github.com/iterative/datachain/pull/514